### DISCUSSION TITLE
Project Management

### DISCUSSION THREAD

**Discussion by gnodar01** [bilayer-containers/bilayers#2] [Announcements]
Generally, we've decided to focus all project management activities on the *bilayer-containers* GitHub org.

This is subject to change (via this discussion thread), but generally we'll have this workflow (colab-flow?):
- Create a discussion to brainstorm ideas, and come to a consensus on a concrete plan for how to implement the idea
- Transfer the discussion into an issue
  - Not all discussions should necessarily become issues; some can just remain ongoing discussions
  - Issues other than minor bug fixes and the like, should start out as discussions
  - We can also create a new issue, and "reference" a comment in an existing discussion, rather than transferring the whole thing
- Create a new project, in the Projects tab, if necessary
  - Projects are just collections of like issues
- Link relevant projects to milestones, or create a new milestone if necessary
- Create a new branch
  - Branches should more or less be focused on one issue at a time, although its perfectly okay to occasionally have a branch which is meant to resolve multiple issues, once completed
  - Branches should not be so big that they are meant to resolve issues across two or more projects
- As soon as the first commit is made on a branch, push it, create a PR, and if you expect to make more commits put the PR in "draft" status
  - In general push commits often, even if they are works in progress; i.e. do not wait until the PR is "completed" before pushing all changes
- Make sure the PR is linked to all issues, projects, milestones, etc
  - A commit that has "resolves #x" in the body should automatically link a issue and close it once the PR is merged, but double check
- Once a PR is ready (once commits have been pushed that fully resolve the linked issues), add a reviewer
- Reviewer will approve, or suggest changes
- Once Approved, merge the PR (actually rebase, as we've decided to try to keep a linear history)

I'm sure exceptions will occur to the above schema, but when they do we should revisit and incorporate any necessary changes, rather than abandoning this structure altogether
My preliminary inclination is that projects should have the following statuses: "To Do", "In Progress", "Blocked", "Done". Feel free to suggest alternatives.

@rajavishah @bethac07 Thoughts?

**Comment by bethac07** [comment]
💯💯💯💯💯

**Comment by gnodar01** [comment]
Commits should have informative titles and messages. See our [contribute guide](https://github.com/piximi/piximi/wiki/contribute#commit-messages) on how to structure commit messages in the Piximi wiki.

If you have a run of commits where you have "fix", "really fix", "ok actually fix this time", that's okay (push it anyway), but then make sure to go back and squash them into one single commit, with an informative message on what was broken and is now fixed.

**Reply by bethac07** [reply]
>If you have a run of commits where you have "fix", "really fix", "ok actually fix this time"

I feel seen
